lib/logstash/inputs/kinesis/worker.rb

```diff
@@ -16,22 +16,22 @@ def initialize(*args)
       @next_checkpoint = Time.now - 600
       @constructed = true
     else
-      _shard_id, _ = args
+      _shard_id = args[0].shardId
       @decoder = java.nio.charset::Charset.forName("UTF-8").newDecoder()
     end
   end
   public :initialize
 
-  def processRecords(records, checkpointer)
-    records.each { |record| process_record(record) }
+  def processRecords(records_input)
+    records_input.records.each { |record| process_record(record) }
     if Time.now >= @next_checkpoint
-      checkpoint(checkpointer)
+      checkpoint(records_input.checkpointer)
       @next_checkpoint = Time.now + @checkpoint_interval
     end
   end
 
-  def shutdown(checkpointer, reason)
-    if reason == com.amazonaws.services.kinesis.clientlibrary.types::ShutdownReason::TERMINATE
+  def shutdown(shutdown_input)
+    if shutdown_input.reason == com.amazonaws.services.kinesis.clientlibrary.types::ShutdownReason::TERMINATE
       checkpoint(checkpointer)
```

Comment:
yes! You are right. I've already pushed that fix.

I think this method should be tested somehow.